### ISSUE TITLE
Enable separeate preview modes for Search & Replace results

### ISSUE THREAD

**Issue by dsanz006** [microsoft/vscode#60131]
There is a setting called `"workbench.editor.enablePreview"`, which I currently have set to `false`, mainly because I use my keyboard arrows to navigate through files in the file explorer, and it annoys me that when I open a file pressing Enter, it opens it in preview mode (I would have to double click on it with my mouse to open it and keep it, but as I said I am barely using my mouse).

However, when I am watching my Search & Replace results, I don't like that every file with a search result opens when I navigate through it. After I finish my search, I end up with a bunch of open editors that I am not going to use anymore. Here is where I would like to use the preview mode.

To summarize, I would like to have the preview modes separately for each context: something like  `"workbench.editor.enablePreviewFromExplorer"` and `"workbench.editor.enablePreviewFromSearch"`. There is currently a `"workbench.editor.enablePreviewFromQuickOpen"`, which I find very useful.

**Comment by skprabhanjan** [comment]
@roblourens , can I work on this ? :)

**Comment by roblourens** [comment]
Honestly that many settings feels like overkill. I see you opened https://github.com/Microsoft/vscode/issues/60027 which I like more, is that enough to address this?

**Comment by dsanz006** [comment]
I opened this issue because I think it is a different matter than #60027. I don't think it is overkill, in my personal opinion a more customizable editor is always better and can adapt better to user needs.

**Comment by roblourens** [comment]
I think you have to find the right balance between customizability and an overload of settings that target micro-scenarios that most users aren't worried about. I will keep it open to collect any other feedback but I'm not convinced.

**Comment by dsanz006** [comment]
I understand, I will wait then. In the meantime, if I enable preview mode, is there any way that I can open the files from the explorer view in non-preview mode without using the mouse?

**Comment by roblourens** [comment]
There is a default keybinding of `cmd+k enter` for "Keep Editor"

**Comment by dsanz006** [comment]
Didn't see that, thank you!

**Comment by dsanz006** [comment]
I tried the "Keep Editor" command but it only works once the editor is opened, there is no such thing like "open and keep editor". That is why I just opened issue #60276. If #60276 feature is added, then this feature can be closed I think, because one can always choose how to open a file.

**Comment by eyalroth** [comment]
This is one of the first things I noticed when working with this editor. I really dislike the preview feature when navigating the explorer, but I find it extremely helpful with the search results.

I'll even go further and suggest -- perhaps not for the scope of this issue -- that the preview should not open as a separate editor, as it is extremely confusing to have a "preview editor" (the italic font isn't a good indicator), but rather should appear as part of the search view.

For example, this is how it works in IntelliJ:

![intellij-search1](https://user-images.githubusercontent.com/6641808/59177162-dbf80b00-8b63-11e9-8e1a-6e1bc319e3a0.png)
(hover search view)

![intellij-search2](https://user-images.githubusercontent.com/6641808/59177180-f3cf8f00-8b63-11e9-9242-d380c2600ed6.png)
(fixed search view)

**Comment by roblourens** [comment]
Ok, I would take a PR to add a setting `workbench.editor.enablePreviewFromSearch`

**Comment by roblourens** [comment]
Discussing in the PR, I think I misunderstood the original request. I realized that the original request was for the search setting to be enabled to override the main setting. But the existing quick open setting doesn't work that way.

Now I am doubting the usefulness of the existing quick open setting since it only works in one direction. And I'm not sure I want to add a similar setting for search if it doesn't actually address the original issue. And there are a ton of ways to open files, and I don't want to add settings for every single one.

@bpasero is there already an issue for more granular enablePreview settings or is this out of scope?

**Comment by bpasero** [comment]
I think the intent is to have a setting that clicking on a search result when in replace mode uses preview mode, while otherwise preview is disabled. I would think this issue did not get a lot of attention, so it would probably not qualify to be considered.

**Comment by roblourens** [comment]
I agree, sorry @kwshoemaker2
